pyvlx/api/leave_learn_state.py

```diff
@@ -1,28 +1,33 @@
 """Module for handling the login to API."""
+from typing import TYPE_CHECKING
+
 from pyvlx.dataobjects import DtoLeaveLearnState
 
 from .api_event import ApiEvent
 from .frames import (
-    FrameLeaveLearnStateConfirmation, FrameLeaveLearnStateRequest)
+    FrameBase, FrameLeaveLearnStateConfirmation, FrameLeaveLearnStateRequest)
+
+if TYPE_CHECKING:
+    from pyvlx import PyVLX
 
 
 class LeaveLearnState(ApiEvent):
     """Class for handling leave learn state to API."""
 
-    def __init__(self, pyvlx):
+    def __init__(self, pyvlx: "PyVLX"):
         """Initialize leave learn state class."""
         super().__init__(pyvlx=pyvlx)
         self.status = DtoLeaveLearnState()
         self.success = False
 
-    async def handle_frame(self, frame):
+    async def handle_frame(self, frame: FrameBase) -> bool:
         """Handle incoming API frame, return True if this was the expected frame."""
         if not isinstance(frame, FrameLeaveLearnStateConfirmation):
             return False
-        self.status = frame.status
+        self.status.status = frame.status
```

Comment:
The previous types were totally different, not sure if this worked before?

Reply:
unclear, it probably never worked.